twitter/direct_messages.go

```diff
@@ -14,3 +19,119 @@ type DirectMessage struct {
+	ID int64 `url:"id,omitempty"`
+}
+
+// Show returns the requested DirectMessage and all response messages.
+// https://dev.twitter.com/rest/reference/get/direct_messages/show
+func (s *DirectMessageService) Show(id int64) ([]DirectMessage, *http.Response, error) {
+	params := &StatusShowParams{ID: id}
+	dms := new([]DirectMessage)
+	apiError := new(APIError)
+	resp, err := s.sling.New().Get("show.json").QueryStruct(params).Receive(dms, apiError)
+	return *dms, resp, relevantError(err, *apiError)
+}
+
+// DirectMessageGetParams are the parameters for DirectMessageService.Get
+type DirectMessageGetParams struct {
+	SinceID         int64 `url:"since_id,omitempty"`
+	MaxID           int64 `url:"max_id,omitempty"`
+	Count           int   `url:"count,omitempty"`
+	IncludeEntities *bool `url:"include_entities,omitempty"`
+	SkipStatus      *bool `url:"skip_status,omitempty"`
+}
+
+// Get returns the all the direct messages
+// https://dev.twitter.com/rest/reference/get/direct_messages
+func (s *DirectMessageService) Get(params *DirectMessageGetParams) ([]DirectMessage, *http.Response, error) {
+	if params == nil {
+		params = &DirectMessageGetParams{}
+	}
+	dms := new([]DirectMessage)
+	apiError := new(APIError)
+	resp, err := s.baseSling.New().Get("direct_messages.json").QueryStruct(params).Receive(dms, apiError)
+	return *dms, resp, relevantError(err, *apiError)
+}
+
+// DirectMessageSentParams are the parameters for DirectMessageService.Sent
+type DirectMessageSentParams struct {
+	SinceID         int64 `url:"since_id,omitempty"`
+	MaxID           int64 `url:"max_id,omitempty"`
+	Count           int   `url:"count,omitempty"`
+	Page            int   `url:"page,omitempty"`
+	IncludeEntities *bool `url:"include_entities,omitempty"`
+}
+
+// Sent returns the all the direct messages sent
+// https://dev.twitter.com/rest/reference/get/direct_messages/sent
+func (s *DirectMessageService) Sent(params *DirectMessageSentParams) ([]DirectMessage, *http.Response, error) {
+	if params == nil {
+		params = &DirectMessageSentParams{}
+	}
+	dms := new([]DirectMessage)
+	apiError := new(APIError)
+	resp, err := s.sling.New().Get("sent.json").QueryStruct(params).Receive(dms, apiError)
+	return *dms, resp, relevantError(err, *apiError)
+}
+
+// DirectMessageNewParams are the parameters for DirectMessageService.New
+type DirectMessageNewParams struct {
+	UserID     int64  `url:"user_id,omitempty"`
+	ScreenName string `url:"screen_name,omitempty"`
+	Text       string `url:"text"`
+}
+
+// New creates a new direct message
+// https://dev.twitter.com/rest/reference/get/direct_messages/new
+func (s *DirectMessageService) New(params DirectMessageNewParams) (DirectMessage, *http.Response, error) {
+	dm := new(DirectMessage)
+	apiError := new(APIError)
+	resp, err := s.sling.New().Post("new.json").BodyForm(params).Receive(dm, apiError)
+	return *dm, resp, relevantError(err, *apiError)
+}
+
+// SendToID sends a direct message by twitter user id
```

Comment:
If these `SendToID` and `SendToScreenName` convenience methods were added, they should take params so they can accommodate changes to the DM service (e.g. if group DMs get exposed). With that parameter though, they don't seem much more valuable than just calling `New` directly.